lib/service_now_request.rb

```diff
@@ -0,0 +1,46 @@
+require 'base64'
+require 'json'
+require 'net/http'
+require 'openssl'
+
+# This class organizes ruby http code used by all of the tasks.
+class ServiceNowRequest
+  def initialize(uri, http_verb, fields, user, password)
+    @uri = URI.parse(uri)
+    @http_verb = http_verb.capitalize
+    @fields = fields
```

Comment:
I think here, `fields` => `body`. `ServiceNowRequest` is more general and could apply to requests outside of the table API.

Reply:
Oh also, should this be in the module's `files` directory? Asking b/c that's where the task helper module puts its code as well.

Reply:
I think according to this [doc](https://puppet.com/docs/puppet/6.10/modules_fundamentals.html#concept-1234) lib sounds slighty better, but I'm open to changing if you disagree.

Reply:
Change `fields` to `body` for the ServiceNowRequest class.

Reply:
Ohh my bad, I just noticed this is in `lib`. No problem, thanks.